src/emu/http.cpp

```diff
@@ -7,6 +7,10 @@ http.cpp
 HTTP server handling
 
 ***************************************************************************/
+#ifdef __sun
+#define ASIO_DISABLE_DEV_POLL
+#define ASIO_HAS_EPOLL
+#endif
```

Comment:
I get a C4603 warning in VS2017 if this is defined before including `emu.h`. @despair86 can it be safely moved below the includes?

Reply:
uhh sure, I'll format a patch in a bit, I sorta left my patches to BX hanging too

Reply:
well ain't that something?

wonder if recent version of Microsoft C++ does weird things with its PCH passes

PCH on literally any other C++ compilation system works fine (I'm a Windows dev by day, but have largely migrated to mingw-w64 for performance and wider platform compat)

*shrugs*

Reply:
Given the way we handle precompiled headers, it's pretty important not to do anything significant before including `emu.h`.  MSVC will give a warning and potentially behave strangely if you do anything beforehand, and with gcc/clang you'll potentially get different behaviour with/without precompiled header enabled, because we supply the PCH as an implicit prefix behaviour.  (We're not using a PCH cache which would just cause the PCH not to be used if preprocessor macro definitions don't match what it was precompiled with at the point of inclusion.)